### PR TITLE
Optimize various rest utility functions

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:
This PR optimizes various rest utility functions to reduce `/api/v1/transactions` cpu overhead.

- Use bitwise operation for entity id encoding / decoding
- Use string operations in `nsToSecNs`
- Use string conversion in `getNullableNumber`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
The performance gain is surprising. Tested with local postgres db,  the single process nodejs RPS is now 104 v.s. 70.

**Checklist**

- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-rest/utils.js [hunk 1]
```diff
@@ -579,11 +579,20 @@ const getPaginationLink = (req, isEnd, field, lastValue, order) => {
 
 /**
  * Converts nanoseconds since epoch to seconds.nnnnnnnnn format
+ *
  * @param {String} ns Nanoseconds since epoch
+ * @param {String} sep separator between seconds and nanso, default is '.'
```
Comment (xin-hedera) [inline hedera-mirror-rest/utils.js:584]:
```suggestion
 * @param {String} sep separator between seconds and nanos, default is '.'
```

File: hedera-mirror-rest/entityId.js [hunk 2]
```diff
@@ -41,7 +45,7 @@ class EntityId {
   getEncodedId() {
     return this.num === null
       ? null
-      : (BigInt(this.num) + BigInt(this.realm) * realmOffset + BigInt(this.shard) * shardOffset).toString();
+      : ((BigInt(this.shard) << shardOffset) | (BigInt(this.realm) << numBits) | BigInt(this.num)).toString();
```
Comment (steven-sheehy) [inline hedera-mirror-rest/entityId.js:48]: Shouldn't these already be BigInt if they are constructed from a BigInt(id) and bitwise BigInt operators in `fromEncodedId()`?

**Review by steven-sheehy — APPROVED** [review]
LGTM with one question

**Review by Nana-EC — DISMISSED** [review]

File: hedera-mirror-rest/entityId.js (hunk 2 above)
Comment (xin-hedera) [reply hedera-mirror-rest/entityId.js:48]: we have unit tests using js `Number` and call `EntityId.of(10, 10, 10)`.

I think moving forward we should consider converting to typescript for strict type check and that'll require more effort.

Comment (steven-sheehy) [reply hedera-mirror-rest/entityId.js:48]: It might be worth converting those tests to BigInt to avoid the runtime performance of converting to BigInt.

**Review by ijungmann — DISMISSED** [review]

File: hedera-mirror-rest/entityId.js (hunk 2 above)
Comment (xin-hedera) [reply hedera-mirror-rest/entityId.js]: made some changes, also added jsdoc to indicate the types of the shard, realm, num for `of` should be `BigInt`

**Review by steven-sheehy — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]

**Review by ijungmann — APPROVED** [review]

**Pull request merged**